Differentiate base paths in repository integration tests

tlrx opened the pull request:
This commit change the repositories base paths used in Azure integration tests so that they don't conflict with each other when tests run in parallel on real Azure storage service.

Closes #47202

tlrx committed:
Revert "Mute Snapshot/Restore with repository-azure (#47204)"

This reverts commit 77fa6cf8c85f7fabf6bb7d613d6b3604a9d9cfd8.

tlrx committed:
Revert "Mute azure third party tests (#47205)"

This reverts commit f487495e022ed4e0e053ebe399b67e3081a831cc.

tlrx committed:
Differentiate repository base paths in Azure tests

elasticmachine commented:
Pinging @elastic/es-distributed

original-brownbear commented on plugins/repository-azure/build.gradle:
```diff
@@ -88,7 +88,7 @@ task thirdPartyTest(type: Test) {
   systemProperty 'test.azure.sas_token', azureSasToken ? azureSasToken : ""
   systemProperty 'test.azure.container', azureContainer ? azureContainer : ""
-  systemProperty 'test.azure.base', azureBasePath ? azureBasePath : ""
+  systemProperty 'test.azure.base', (azureBasePath ? azureBasePath : "") + "_third_party_tests_" + project.testSeed
```
@tlrx thanks for this one!

I think technically this is the perfect fix in terms of stability. One problem we're creating here though is that we are leaking a bunch of blobs for each test run now.
Should we add some logic to delete the leaked folders maybe? (otherwise we force infra to deal with this somehow which will be tricky for them I think)

original-brownbear replied:
To avoid this issue for now: Maybe we can simply add a constant path here and another constant in the other build so we don't get interference between the test runs? We don't have to worry about different builds interfering with each other, infra makes sure that doesn't happen so that should be fine?

tlrx replied:
Yeah, I wrongly assumed that the container/buckets were recreated on demand by some Infra task before running the tests.

I think we can easily keep the base_path with test seed for the 3rd party tests, as we can easily clean up the files when tearing down the test (similarly to what the test set up is already doing doing)

For the repository integration test, I'll go with adding a constant.

tlrx committed:
clean up afterwards in 3rd party tests

original-brownbear reviewed (approved):
LGTM (assuming CI is happy too) :) Its retriever the 3rd party tests right away after merging this to make sure we're good here.

original-brownbear reviewed (changes requested):
Seems CI is unhappy, that teardown doesn't work with HDFS repo for some reason.

tlrx committed:
S3+GCS

tlrx committed:
Adapt for HDFS

tlrx commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into azure-it

tlrx commented:
@original-brownbear I've updated to code to fix the HDFS test issue. I've also added changes for S3/GCS (will update title+desc when merging) alongside. Can you please have another look?

original-brownbear reviewed (approved):
LGTM (if CI loves it :D), thanks @tlrx !

tlrx commented:
Thanks Armin!

romseygeek commented:
This just failed on 6.8 as well, can it be backported there?

https://gradle-enterprise.elastic.co/s/pdvqg6hefhjd6/

tlrx commented:
Thanks @romseygeek - I opened #48286
